lib/models/addon.js

```diff
@@ -154,7 +155,13 @@ Addon.prototype._unwatchedTreeGenerator = function unwatchedTree(dir) {
   @return {Boolean}
 */
 Addon.prototype.isDevelopingAddon = function() {
-  return process.env.EMBER_ADDON_ENV === 'development';
+  if (process.env.EMBER_ADDON_ENV === 'development' && (this.parent instanceof Project)) {
+    var addon = this.addonDiscovery.discoverFromProjectItself(this.parent);
```

Comment:
Yeah that's why I'm checking `this.parent instanceof Project`, the `project` param is marked as deprecated at the top of the addon, so I wasn't sure if I could reliably use it.